apps/dotcom/client/e2e/tests/homepage.spec.ts

```diff
@@ -1,21 +1,66 @@
+import { sleep } from 'tldraw'
 import { expect, test } from '../fixtures/tla-test'
 
 test.beforeEach(async ({ homePage }) => {
 	await homePage.goto()
 	await homePage.isLoaded()
 })
 
+test.afterEach(async ({ database }) => {
+	await database.reset()
+})
+
 test('can toggle sidebar', async ({ editor, sidebar }) => {
 	await editor.ensureSidebarClosed()
 	await expect(sidebar.sidebarLogo).not.toBeVisible()
 	await editor.toggleSidebar()
 	await expect(sidebar.sidebarLogo).toBeVisible()
 })
 
-test('can create new document', async ({ page, editor, sidebar }) => {
+test('can create new document', async ({ editor, sidebar }) => {
 	await editor.ensureSidebarOpen()
-	const currentCount = (await page.$$('[data-element="file-link"]')).length
+	const currentCount = await sidebar.getNumberOfFiles()
 	await sidebar.createNewDocument()
-	const newCount = (await page.$$('[data-element="file-link"]')).length
+	const newCount = await sidebar.getNumberOfFiles()
 	expect(newCount).toBe(currentCount + 1)
 })
+
+test('sidebar file operations', async ({ page, sidebar, deleteFileDialog }) => {
+	const randomString = Math.random().toString(36).substring(7)
+
+	await test.step('rename the document via double click', async () => {
+		const fileLink = await sidebar.getFileLink()
+		await fileLink?.dblclick()
+		const input = page.getByRole('textbox')
+		await input?.fill(randomString)
+		await page.keyboard.press('Enter')
+		const newFileLink = await sidebar.getFileLink()
+		const newName = await newFileLink?.innerText()
+		expect(newName).toBe(randomString)
+	})
+
+	await test.step('duplicate the document', async () => {
+		const fileName = await sidebar.getFileLink()
+		await sidebar.openFileMenu(fileName)
+		await page.getByRole('menuitem', { name: 'Duplicate' }).click()
+		await sleep(1000)
+		const copiedFileLink = page.getByRole('link').getByText(`${fileName} Copy`)
```

Comment:
I'm expecting to get the link here, but the text is in nested inside a sibling div so this isn't returning anything. I think the text should be moved inside the link so I've started a minor refactor of the TlaSidebarFileLink component.